static/js/src/advantage/react/components/Subscriptions/SubscriptionList/ListGroup/ListGroup.tsx

```diff
@@ -1,27 +1,35 @@
-import { ContextualMenu } from "@canonical/react-components";
-import React from "react";
+import React, { useRef, useState } from "react";
 import type { ReactNode } from "react";
 
+import RenewalSettings from "../RenewalSettings";
+
 type Props = {
   children: ReactNode;
   title: string;
 };
 
-const ListGroup = ({ children, title }: Props): JSX.Element => (
-  <div className="p-subscriptions__list-group">
-    <div className="p-subscriptions__list-group-title">
-      <span className="p-text--x-small-capitalised u-align-text--small-to-default u-no-margin--bottom">
-        {title}
-      </span>
-      <ContextualMenu
-        hasToggleIcon
-        links={[]}
-        toggleClassName="is-dense u-no-margin--bottom"
-        toggleLabel="Renewal settings"
-      />
+const ListGroup = ({ children, title }: Props): JSX.Element => {
+  const positionNode = useRef<HTMLDivElement | null>(null);
+  const [, setRefReady] = useState(false);
+  return (
+    <div className="p-subscriptions__list-group">
+      <div
+        className="p-subscriptions__list-group-title"
+        ref={(element) => {
+          positionNode.current = element;
+          // Fire state change so that the menu rerenders now that the ref
+          // exists in the DOM.
+          setRefReady(true);
+        }}
```

Comment:
Can you explain what's going on here?

Reply:
Yeah, so because React uses a shadow DOM, when we set up a ref e.g. `ref={positionNode}` there is no way to know when that ref exists in the real DOM (refs by their nature don't cause rerenders). But because you usually use refs to interact with DOM elements React has provided a pattern called ["Callback refs"](https://reactjs.org/docs/refs-and-the-dom.html#callback-refs) so that you can do something when the element gets rendered.

In our case we set the current ref to the element and because we want to the contextual menu to update it's position/size from that element we also do a state update (`setRefReady`) to cause the components to rerender now that the position node exists.

Does that make sense?

Reply:
Oh ok, I didn't know about the callback ref, thanks for the explanation!